### ISSUE TITLE
[BUG] Test for String.find falsely passes

### ISSUE THREAD

**Issue by liz3** [dictu-lang/Dictu#748]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

This testcase:
```
this.assertEquals("Dictu is great! Dictu is great! Dictu is great!".find("Dictu", 3), 37);
```
Passes in the current builds.

### Expected Behavior

I was a bit confused because this test failed over and over in my current endavours of hacking on utf8 support, until i looked somewhat closer at the test:
```
this.assertEquals("Dictu is great! Dictu is great! Dictu is great!".find("Dictu", 3), 37);
```
The third occurance of `Dictu` is at index 32, not at 37= index + search length!
So this seams to be a bug? in the current implementation of find?

### Steps To Reproduce

Run the string tests for find:
`dictu tests/strings/find.du`

### Anything else?

Current test reference: https://github.com/dictu-lang/Dictu/blob/develop/tests/strings/find.du#L24

**Comment by liz3** [comment]
@Jason2605 I wanted to bump this

**Comment by Jason2605** [comment]
Sorry somehow missed this (emails get buried sometimes, need to be more vigilant checking this repo, apologies). Oh wow, that's really bad from me, yeah this is an issue

**Comment by Jason2605** [comment]
Fixed as part of #749
